addons/website_sale/models/product_template.py

```diff
@@ -440,6 +440,9 @@ def _get_combination_info(
                 combination_info,
             )
 
+        if product_or_template.type == 'combo' and website.show_line_subtotals_tax_selection == 'tax_included':
+            combination_info['tax_disclaimer'] = _("Final price may vary based on selection. Tax will be calculated at checkout.")
```

Comment:
```suggestion
        if (
            product_or_template.type == 'combo'
            and website.show_line_subtotals_tax_selection == 'tax_included'
            and not all(
                tax.price_include
                for tax in product_or_template.combo_ids.combo_items_ids.product_id.taxes_id
            )
        ):
            combination_info['tax_disclaimer'] = _(
                "Final price may vary based on selection. Tax will be calculated at checkout."
            )
```
Lines should be kept 100 chars max, plus tax calculations should already be correct if the taxes used are configured to be included in the price.